### PR TITLE
Added remaining layer lock handling

### LINKED ISSUE #11428: Add remaining layer lock functionality
## Feature Description

See upcoming PR for full list of features to be added.

### PR DESCRIPTION
**Pull request by barklund**
## Context

This adds a bunch of extra features related to locked layers. Hopefully everything to make the feature fully functional.

See todo-list for details.

## Technical choices
Some simplifications and assumptions for technical reasons:
* Multi-selection is not possible if any element is locked. This makes multi-selection transformations easier to handle (they're always allowed because no element in selection can be locked)
* You cannot right-click a locked element directly, but once selected, you can. You can however no longer "click through" a locked element once selected.

Both of the above scenarios deviate from Figma's handling of locked elements, but both of the above solutions were much easier to implement than to copy how Figma does them.

## To-do

* [x] When dragging media, do not highlight locked media and shape as potential drop target nor allow media to be dropped "into" locked elements
* [x] When hovering a locked element, do not display a slight blue border
* [x] Do not allow locked elements to be clicked, unless already selected (can be selected through right-click menu or layer panel)
* [x] Filter out locked elements from lasso selection
* [x] Filter out locked elements from shift-selection in layer panel
* [x] When meta-selecting a non-locked layer via layer panel, if old selection was a locked layer, only select new element
* [x] When meta-selecting a locked layer via layer panel, forget old selection and only select the locked layer
* [x] When shift-selecting a non-locked layer via canvas, if old selection was a locked layer, only select new element
* [x] Do not show links when hovering locked elements
* [x] When a locked element is selected, do not allow any transformation of the element using moveable (hide all handles and use default cursors)
* [x] When a locked media or shape element is selected, allow right-click to open context menu
* [x] When a locked non-media/non-shape element is selected, allow right-click to open context menu
* [x] When a locked element is selected, do not allow movement via keyboard arrows
* [x] When focusing a locked element on canvas, make lock status part of element accessible name
* [x] When editing a locked media element, do not allow resizing the frame
* [x] When focusing a locked element, announce lock state as part of element name for screen-readers.
* [x] When a locked text element is selected, do not allow edit mode using enter key, any key, or click inside
* [x] Add unit tests for changes to story reducer
* [x] Add karma tests for changes to layer selection
* [x] Add karma tests for changes to lasso selection
* [x] Add karma tests for changes to canvas selection

## User-facing changes

_TODO_

## Testing Instructions

This PR can be tested by following these steps:

1. Add multiple elements to canvas of different types
2. Stack some on top of each other
3. Lock some layers, especially some of the top ones
4. Observe that layer and element selection makes sense both using canvas, lasso, and layer panel.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11428

### CODE REVIEW

File: packages/story-editor/src/components/canvas/frameElement.js [hunk 1]
```diff
@@ -269,6 +283,18 @@ function FrameElement({ id }) {
...
+  const lockedElementText = __('Locked element: %s', 'web-stories');
+
+  // translators: %s: Name of element
+  const unlockedElementText = __('Element: %s', 'web-stories');
+
+  const layerText = getLayerText(element);
+  const elementText = element.isLocked
+    ? lockedElementText
+    : unlockedElementText;
+  const elementLabel = sprintf(elementText, layerText);
```
Comment (swissspidy) [inline packages/story-editor/src/components/canvas/frameElement.js]: To address the i18n warning:

```suggestion
  const layerText = getLayerText(element);
  const elementLabel = element.isLocked
    ? sprintf(
        // translators: %s: Name of element
        __('Locked element: %s', 'web-stories'),
        layerText
      )
    : sprintf(
        // translators: %s: Name of element
        __('Element: %s', 'web-stories'),
        layerText
      );
```

Comment (barklund) [reply packages/story-editor/src/components/canvas/frameElement.js]: Thanks!

**Review by merapi — DISMISSED** [review]
This PR looks good. Not sure if this bug is related (but I can repro it here, not on main). Maybe it will be self-resolved with the 2nd Locking PR:

https://user-images.githubusercontent.com/21036224/167850678-2f09a4ae-7c65-4c30-841c-558425be60fa.mov

File: packages/story-editor/src/app/rightClickMenu/useLayerSelect.js [hunk 2]
```diff
@@ -136,15 +136,17 @@ function useLayerSelect({ menuItemProps, menuPosition, isMenuOpen }) {
...
-        label: isBackground ? (
-          <span>{__('Background', 'web-stories')}</span>
-        ) : (
-          <LayerContent element={element} />
+        label: (
+          <span>
+            {isBackground
+              ? __('Background', 'web-stories')
+              : getLayerText(element)}
+          </span>
```
Comment (barklund) [inline packages/story-editor/src/app/rightClickMenu/useLayerSelect.js]: @merapi, [it](https://github.com/GoogleForCreators/web-stories-wp/pull/11429#pullrequestreview-969268945) was actually a bug in the right-click menu, as it used the element `LayerContent` property, that I had deprecated. Good find, fixed now.

**Review by merapi — DISMISSED** [review]
This does the job of layer locking really well (excluding some stuff like right-clicking to unlock).

I was thinking about adding "Lock/Unlock" to RCM so you can select a bunch of elements on the canvas and lock them, right now you need to go one by one - can be done in v2.

**Review by merapi — APPROVED** [review]

**Pull request merged**